evm/p2p/kademlia.py

```diff
@@ -264,6 +268,20 @@ def neighbours(self, node_id, k=k_bucket_size):
         return sort_by_distance(nodes, node_id)[:k]
 
 
+def binary_get_bucket_for_node(buckets, node):
+    """Return the bucket for a given node."""
+    sorted_buckets = sorted(buckets)
+    bucket_ends = list(KBucket.end for KBucket in sorted_buckets)
```

Comment:
canonical version of this is just to do it as a list comprehension:

```python
bucket_ends = [bucket.end for bucket in sorted_buckets]
```

Though I like tuples since they are immutable (surprise surprise!) so I'd be fine with this too.

```python
bucket_ends = tuple(bucket.end for bucket in sorted_buckets)
```